### DISCUSSION TITLE
Any plan or route to add this to main Java instrumentation with default turned off?

### DISCUSSION THREAD

**Discussion by tom-diacono** [DGuhr/spring-otel-extended#2] [General]
Hi @DGuhr ,

I really like the look of this project. If I'm honest- it's what I was hoping to get out of OpenTelemetry from the get go. Having said which, I'm not sure I could use it in prod for now (in your own words, may not be prod ready).

Given that, do you think there's any scope for this to be added to the official Spring instrumetation- defaulted to being 'turned off'? I'd be really surprised if I was the only one out there who was after something like this with slightly more 'official support'.

I'm not sure how contributions happen- whether the guys who maintain the Spring instrumentation would be able to pick this up or what?

Anyhow, really cool project, thanks for building it

**Comment by DGuhr** [comment]
Hey @tom-diacono , first of all thanks for the kind words. :) 

I just asked in the CNCF slack if someone more official than me might want to answer, but to be honest I don't think there'll be much traction into integrating this (or something like this) in the official spring auto instrumentation, for the reasons I lined out in the readme. 

But hey, let's see how it goes. Here's the question, fwiw: [link](https://cloud-native.slack.com/archives/C014L2KCTE3/p1718892664820429) (you have to be [registered](https://communityinviter.com/apps/cloud-native/cncf) to the cncf slack, though)

**Reply by tom-diacono** [reply]
@DGuhr  Nice- I've just joined CNCF slack so will keep an eye out if anyone starts a thread.

Thanks
